nanoFramework.Azure.Devices.Client/DeviceClient.cs

```diff
@@ -28,17 +30,25 @@ public class DeviceClient : IDisposable
         private readonly string _deviceId;
         private readonly string _sasKey;
         private readonly string _telemetryTopic;
-        private readonly X509Certificate2 _clientCert;
+        
         private readonly string _deviceMessageTopic;
         private Twin _twin;
         private bool _twinReceived;
+#if FULLYMANAGED
```

Comment:
Maybe we could inject mqtt client in constructor via interface instead?

Reply:
yes, I was thinking of that. I've been hesitating a lot.
I've been hesitating to pass it in the Open function or the constructor.
And finally the reason I decided to pass it in the Open function and expose it only in the case of the full managed lib is that it was actually being created here for the normal lib.
I'm open to get the pro and cons and more what you have in mind!

Reply:
Injecting in constructor makes it easy to use with DI framework (just add into container and create object via ioc). That's the only reason. But it will require major refactor for this class.

Reply:
The core idea is to keep the class as compatible as possible with the big .NET API. And not breaking change with the existing one. What I can do is add a constructor which will take the IMqttClient rather than passing it on the Open function. That may indeed be a bit cleaner.